### PR TITLE
Rework Serialization

### LINKED ISSUE #182: Panic when interrupting a long computation
Error is
```rs
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: SendError(Ok(FetchChunk { body: Some(Data([14, 0, 0, 0, 0, 0, 1, 142, 150, 14, 0, 0, 0, 0, 0, 1, 143, 150, 14, 0, 0, 0, 0, 0, 1, 144, 150, 14, 0, 0, 0, 0])) }))', bastionlab_polars/src/serialization.rs:134:14
```

Reproduction is: run a long computation, and interrupt the python script, this error pops up in bastionlab console

This is the result of an unwrap() that I will probably get to cleanup soon

### PR DESCRIPTION
**Pull request by cchudant**
Here are the details:
- Serialization is now using Arrow IPC (feather) as a data format
- It is using the File (randomaccess) version of arrow IPC, there is a Streaming format that is perfect for our usecase, but it's not exposed to the python polars api as of now – the goal would be to switch to this at some point :partying_face: 
  - also, the polars IPCStreamingReader/Writer are problematic since they require the stream to be Seek, and we would have to fix this upstream too
- Tokio channel task is now quite clean
- Hashing of dataframe is still janky, and I would love to see it go away in the future if possible :pray: 
- sanitized_columns is now a `repeated string` (string array) in protobuf, and not json anymore

Closes: #183 
```
rows: 1, time: 0.002671251000265329
rows: 10, time: 0.003582135000215203
rows: 100, time: 0.003072376000091026
rows: 1000, time: 0.0026271389997418737
rows: 10000, time: 0.0033675479999146773
rows: 100000, time: 0.013827925999976287
rows: 1000000, time: 0.1373893780000799
```

Closes #182, because of the changes around the ReceiverStream and the task (see code comments)

### CODE REVIEW

**Comment by cchudant** [comment]
After reviewing with Lucas, I need to:
- [ ] Make sure the CI is passing (spurious error? real issue? idk)
- [ ] I took out some python function comments by accidents, put them back
- [ ] Remove bincode useless dependency

**Comment by cchudant** [comment]
We can merge this once #190 is landed

**Comment by cchudant** [comment]
@dhalf Ready your review & merge :)

**Review by dhalf — APPROVED** [review]

**Pull request merged**
